runtime/worker_sleep.h

```diff
@@ -168,31 +160,17 @@ static bool try_to_disengage_thief(global_state *g, worker_id self,
         cilk_mutex_lock(&g->index_lock);
 
         // Decrement the number of disengaged workers.
-        while (true) {
-            // Atomically decrement the number of disengaged workers.
-            uint64_t disengaged_sentinel = atomic_load_explicit(
-                &g->disengaged_sentinel, memory_order_relaxed);
-            uint32_t disengaged = GET_DISENGAGED(disengaged_sentinel);
-            uint32_t sentinel = GET_SENTINEL(disengaged_sentinel);
-            new_disengaged_sentinel =
-                DISENGAGED_SENTINEL(disengaged - 1, sentinel + 1);
-
-            if (atomic_compare_exchange_strong_explicit(
-                    &g->disengaged_sentinel, &disengaged_sentinel,
-                    new_disengaged_sentinel, memory_order_release,
-                    memory_order_acquire)) {
-                // Update the index structure.
-                last_index = nworkers - GET_DISENGAGED(disengaged_sentinel);
-                if (worker_to_index[self] > last_index) {
-                    swap_worker_with_target(g, self, last_index);
-                }
+        uint64_t disengaged_sentinel =
+            atomic_fetch_add(&g->disengaged_sentinel, 0xffffffff00000001UL);
```

Comment:
Can we avoid using a magic constant here?  Something like `DISENGAGED_SENTINEL(-1, 1)` might work instead of `0xffffffff00000001UL`.